### ISSUE TITLE
A question about equation 4

### ISSUE THREAD

**Issue by peter943** [clovaai/overhaul-distillation#28]
Hi,
Your work is great. I have a little question about equation 4. In your paper, you say: For a negative teacher response, if the student response is higher than the target value, it should be reduced, but if the student response is lower than the target value, it doesn’t need to be increased since negatives are equally blocked by ReLU regardless of their values.

If a student response is negative but higher than the target value, the loss still exists according to equation (4). But the negative student response could be equally blocked by ReLU.  I do not understand the reason to put positive loss in the above scenario. Could you please give me some help? Thanks a lot.

**Comment by bhheo** [comment]
Hi

The reason is not simple but I will do my best to explain it.

At first, distillation try to make teacher response with inferior student. Therefore, distillation loss decrease during training but, the loss can't be zero in general cases. 
Second, gradients of L2 loss decrease when the distance is close to zero.

If we use L2 distance to zero (instead of negative target), student responses will be reduced until zero. but, it doesn't need to be below zero. Because gradient for small positive response is also very small and there are many gradients from other non-zero loss part.
Actually, the zero target value is not good to make negative student response in our experiments.
Therefore, we need small negative target instead of zero to make negative student response.
So, we make a negative margin in eq 2 and use the L2 loss to the margin in eq 3. It is an approximation of L0 loss function only for negative responses.

This topic was discussed in my previous work "Knowledge Transfer via Distillation of Activation Boundaries Formed by Hidden Neurons". So, I didn't include details about it on the paper. I'm sorry for that.

**Comment by peter943** [comment]
Thanks for your patient reply. It is very helpful for me to understand the idea of the paper.
